querydsl-sql/src/main/java/com/querydsl/sql/postgresql/PostgreSQLQuery.java

```diff
@@ -57,8 +57,7 @@ public PostgreSQLQuery(Connection conn, Configuration configuration, QueryMetada
      * @return the current object
      */
     public PostgreSQLQuery<T> forShare() {
-        QueryFlag forShareFlag = configuration.getTemplates().getForShareFlag();
-        return addFlag(forShareFlag);
+        return super.forShare();
```

Comment:
I think at least adding a comment why this is done (the 'superfluous super call') would prevent people from trying to remove it.
So `global forShare support was added later, delegating to super implementation` maybe?